Sources/KlaviyoSwift/Models/Profile.swift

```diff
@@ -0,0 +1,120 @@
+//
+//  Profile.swift
+//
+//
+//  Created by Ajay Subramanya on 8/6/24.
+//
+
+import AnyCodable
+import Foundation
+import KlaviyoCore
+
+public struct Profile: Equatable {
+    public enum ProfileKey: Equatable, Hashable, Codable {
+        case firstName
+        case lastName
+        case address1
+        case address2
+        case title
+        case organization
+        case city
+        case region
+        case country
+        case zip
+        case image
+        case latitude
+        case longitude
+        case custom(customKey: String)
+    }
+
+    public struct Location: Equatable {
+        public var address1: String?
+        public var address2: String?
+        public var city: String?
+        public var country: String?
+        public var latitude: Double?
+        public var longitude: Double?
+        public var region: String?
+        public var zip: String?
+        public var timezone: String?
+
+        /// - Parameters:
+        ///   - address1: First line of street address
+        ///   - address2: Second line of street address
+        ///   - city: city name
+        ///   - country: country name
+        ///   - latitude: Latitude coordinate. We recommend providing a precision of four decimal places.
+        ///   - longitude: Longitude coordinate. We recommend providing a precision of four decimal places.
+        ///   - region: Region within a country, such as state or province
+        ///   - zip: Zip code
+        ///   - timezone: Time zone name. We recommend using time zones from the IANA Time Zone Database.
+        public init(address1: String? = nil,
+                    address2: String? = nil,
+                    city: String? = nil,
+                    country: String? = nil,
+                    latitude: Double? = nil,
+                    longitude: Double? = nil,
+                    region: String? = nil,
+                    zip: String? = nil,
+                    timezone: String? = nil) {
+            self.address1 = address1
+            self.address2 = address2
+            self.city = city
+            self.country = country
+            self.latitude = latitude
+            self.longitude = longitude
+            self.region = region
+            self.zip = zip
+            self.timezone = timezone ?? environment.timeZone()
```

Comment:
minor nit pick push back - would that be a breaking change to the public API? This is unlikely but if someone was passing in nil this would break for them. We may have to do a breaking release if we did this.

Reply:
Ah I see, yeah it's going to be a bit of an adjustment for me to be thinking in terms of external SDK consumers. 

This would be an easy fix for any consumers, because it'll throw a compiler error. But maybe not worth making this change if we don't want to bother them with it